File: app/src/main/java/org/deiverbum/app/data/entity/LHGospelCanticleWithAntiphon.java
package org.deiverbum.app.data.entity;

import androidx.room.Embedded;
import androidx.room.Relation;

import org.deiverbum.app.model.LHGospelCanticle;

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2022.2
 */
public class LHGospelCanticleWithAntiphon {
    @Embedded
    public LHGospelCanticleJoinEntity ce;
    @Relation(
            parentColumn = "antiphonFK",
            entityColumn = "antiphonID",
            entity = AntiphonEntity.class
    )
    public AntiphonEntity antifona;

    public String getAntifona() {
        return antifona.getAntifona();
    }

    public LHGospelCanticle getDomainModel(Integer tipo) {
        LHGospelCanticle ce=new LHGospelCanticle();
        ce.setTipo(tipo);
        ce.setAntifona(getAntifona());
        return ce;
    }
}

File: app/src/main/java/org/deiverbum/app/core/database/model/relation/LHGospelCanticleWithAntiphon.kt
package org.deiverbum.app.core.database.model.relation

import androidx.room.Embedded
import androidx.room.Relation
import org.deiverbum.app.core.database.model.entity.LHAntiphonEntity
import org.deiverbum.app.core.database.model.entity.LHGospelCanticleEntity
import org.deiverbum.app.core.database.model.entity.asExternalModel
import org.deiverbum.app.core.model.data.LHGospelCanticle

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */
data class LHGospelCanticleWithAntiphon(
    @Embedded
    var ce: LHGospelCanticleEntity,

    @Relation(
        parentColumn = "antiphonFK",
        entityColumn = "antiphonID",
        entity = LHAntiphonEntity::class
    )
    var antifona: LHAntiphonEntity
)

fun LHGospelCanticleWithAntiphon.asExternalModel(type: Int): LHGospelCanticle {
    return LHGospelCanticle(type, mutableListOf(antifona.asExternalModel()))
}



